kuma/javascript/src/bcd-signal.jsx

```diff
@@ -0,0 +1,809 @@
+                e.preventDefault();
+                if (window.innerWidth >= 1024) {
+                    return;
+                }
+                const scrollEndElem = scrollTo;
+
+                requestAnimationFrame(timestamp => {
+                    const stamp = timestamp || new Date().getTime();
+                    const duration = 100;
+                    const start = stamp;
+
+                    const startScrollOffset = window.pageYOffset;
+                    const scrollEndElemTop = scrollEndElem.getBoundingClientRect()
+                        .top;
+
+                    scrollToElem(
+                        start,
+                        stamp,
+                        duration,
+                        scrollEndElemTop,
+                        startScrollOffset
+                    );
+                });
+            });
+        }
+
+        return bcSignalCompleteBlock;
+    };
+
+    /**
+     * Scrolls to a specified element
+     */
+    const scrollToElem = (
+        startTime,
+        currentTime,
+        duration,
+        scrollEndElemTop,
+        startScrollOffset
+    ) => {
+        const easeInCubic = t => t * t * t;
+        const runtime = currentTime - startTime;
+        let progress = runtime / duration;
+
+        progress = Math.min(progress, 1);
+
+        const ease = easeInCubic(progress);
+
+        window.scroll(0, startScrollOffset + scrollEndElemTop * ease);
+        if (runtime < duration) {
+            requestAnimationFrame(timestamp => {
+                const currentTime = timestamp || new Date().getTime();
+                scrollToElem(
+                    startTime,
+                    currentTime,
+                    duration,
+                    scrollEndElemTop,
+                    startScrollOffset
+                );
+            });
+        }
+    };
+
+    /**
+     * Creates and returns the signal element HTML
+     * @returns signal element HTML
+     */
+    const signalElem = () => {
+        const extContainer = document.createElement('div');
+        const separator = document.createElement('hr');
+        const container = document.createElement('div');
+        const signalLink = document.createElement('a');
+        signalLink.textContent = gettext('What are we missing?');
+        signalLink.setAttribute('class', 'scroll-to-signal');
+        signalLink.href = '#';
+
+        signalLink.addEventListener('click', function() {
+            toggleBcSignalBlock();
+        });
+
+        container.setAttribute('class', 'signal-link-container');
+        extContainer.setAttribute('class', 'signal-link-ext-container');
+        container.appendChild(separator);
+        container.appendChild(signalLink);
+        extContainer.appendChild(container);
+        return extContainer;
+    };
+
+    bcTable = document && document.querySelector('.bc-table');
+
+    if (!bcSignalsRendered && bcTable && bcTable.insertAdjacentElement) {
```

Comment:
Is this necessary? Haven't you already checked for that on the 2nd line of this function? (referring to the `bcSignalsRendered` boolean variable)